### PR TITLE
ref(ui): Added svgIcon component

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Description:**
Created the component svgIcon.tsx to be used by all our SVG icons components. The idea is that when we want to make a change, for example, in size, we do it only once, in the component svgIcon.tsx

the next steps would be:
- make the color prop, theme's color only
- make the size prop, theme's size only
- remove the viewBox prop and make it fix with the value of `0 0 16 16`. At the moment this is not possible, because the Iconterminal has to have the viewbox of `0 0 471.362 471.362`

### CODE REVIEW

File: src/sentry/static/sentry/app/icons/iconActivity.tsx [hunk 1]
```diff
@@ -1,23 +1,18 @@
...
-      <path d="M15.19,8.74H5.25a.75.75,0,0,1,0-1.5h9.94a.75.75,0,0,1,0,1.5Z" />
-      <path d="M15.19,15H5.25a.75.75,0,1,1,0-1.5h9.94a.75.75,0,0,1,0,1.5Z" />
-      <path d="M15.19,2.53H5.25a.75.75,0,0,1,0-1.5h9.94a.75.75,0,1,1,0,1.5Z" />
-      <path d="M2.25,8.74H.71a.75.75,0,1,1,0-1.5H2.25a.75.75,0,0,1,0,1.5Z" />
-      <path d="M2.25,15H.71a.75.75,0,0,1,0-1.5H2.25a.75.75,0,0,1,0,1.5Z" />
-      <path d="M2.25,2.53H.71A.75.75,0,0,1,.71,1H2.25a.75.75,0,1,1,0,1.5Z" />
-    </svg>
-  );
-});
+const IconActivity = (props: Props) => (
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/icons/iconActivity.tsx:7]: Was the removal of the ref-forwarding intentional?

**Comment by EvanPurkhiser** [comment]
Something I've wanted to fix with these icons, is specify properly whether it has variations of `circle`, `solid` etc in it's types. Right now every icon has the same props, but setting `circle` doesn't always do anything.

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbs/breadcrumbsSearch.tsx [hunk 2]
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 import styled from '@emotion/styled';
 
 import TextField from 'app/components/forms/textField';
-import {IconSearch} from 'app/icons/iconSearch';
+import IconSearch from 'app/icons/iconSearch';
 import {t} from 'app/locale';
-import {IconClose} from 'app/icons/iconClose';
+import IconClose from 'app/icons/iconClose';
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/events/interfaces/breadcrumbs/breadcrumbsSearch.tsx:7]: Should we change all of these icon imports (now that we are touching those files) to 
```
import {IconClose, IconSearch} from 'app/icons';
```
?

File: src/sentry/static/sentry/app/icons/iconActivity.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/icons/iconActivity.tsx:7]: yes...now I have it in the svgIcon 😉

**Comment by priscilawebdev** [comment]
@EvanPurkhiser I also noticed that `circle`, `solid` and `direction` are being used by only a few icons. For this reason, I decided not to add them as prop in the svgIcon component. I believe it makes sense to add `circle`, `solid` and `direction` as props only when all icons make use of them.

**Comment by doralchan** [comment]
@EvanPurkhiser Yeah only some icons need to be circled or be solid. Not sure how to solve for this when it's selective

**Comment by EvanPurkhiser** [comment]
It would be pretty straightforward to have a typescript helper type that you can tell what additional props it supports with a template parameter:

`SvgIcon<'solid' | 'round'>`

File: src/sentry/static/sentry/app/icons/iconActivity.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/icons/iconActivity.tsx:7]: Does that work? Actually didn't think you could spread the ref prop.

**Comment by priscilawebdev** [comment]
@EvanPurkhiser I agree with what you said, but I since not all icons make use of the props `isCircled`, `isSolid` and `direction`.. I would prefer to keep the props only on the icons that use them.

**Comment by EvanPurkhiser** [comment]
Yes that's what I am saying, is to have a helper type that allows you to specify what props the icon will get, something like this: 

```ts
export type SvgIcon<Extra extends 'circle' | 'filled'> = {
  // ...
} & (Extra extends 'circle' ? {isCircle: boolean} : {}) &
  (Extra extends 'filled' ? {isFilled: boolean} : {});

```

**Review by doralchan — APPROVED** [review]
Nice work

File: src/sentry/static/sentry/app/components/events/interfaces/breadcrumbs/breadcrumbsSearch.tsx (hunk 2 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/breadcrumbs/breadcrumbsSearch.tsx:7]: @matejminar after reading this blog post https://humanwhocodes.com/blog/2019/01/stop-using-default-exports-javascript-module/ I understand why you prefer named exports imports over default export and..I agree with your..let's do it! 🤜🤛

File: src/sentry/static/sentry/app/icons/iconActivity.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/icons/iconActivity.tsx:7]: Hmm I have done it before and never had any issue so far. but just to be 100% sure let me do some research... good point

**Comment by priscilawebdev** [comment]
@EvanPurkhiser I really like your idea, but it wouldn't work because the typescript understands that 'circle' is of type string and not "circle" or "solid". Also the unit types don't fit here ... if you don't mind, I'll merge it and if necessary, we can update it later

**Pull request merged**